### PR TITLE
D davidkov/list refactoring

### PR DESCRIPTION
**Pull request by DDavidkov**
#528

IgxListComponent refactoring:

- Remove unnecessary properties/inputs
- Add new ones that are more suited for the component
- Expose new event for item click
- Fix various miscellaneous issues in the component
- Refactored the ListItemComponent and ListComponent into separate files and added a new file for common directive/interfaces

### CODE REVIEW

**Comment by kdinev** [comment]
@DDavidkov Sync your branch and check that the list performance demo still works.

File: src/list/list-item.component.ts [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+        private _renderer: Renderer2) {
+    }
+
+    @Input() public isHeader: boolean;
+
+    @HostBinding("attr.role") public get role() {
+        return this.isHeader ? "separator" : "listitem";
+    }
+
+    @HostBinding("hidden") @Input() public hidden: boolean;
```
Comment (rkaraivanov) [inline src/list/list-item.component.ts]: Format all of the bindings and inputs to follow the format:
```typescript
@Decorator()
method() 
// or
@Decorator()
property
```

File: src/list/list-item.component.ts [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    }
+
+    public get element() {
+        return this.elementRef.nativeElement;
+    }
+
+    public get width() {
+        if (this.element) {
+            return this.element.offsetWidth;
+        } else {
```
Comment (rkaraivanov) [inline src/list/list-item.component.ts]: This seems like a redundant `else`.

File: src/list/list.component.spec.ts [hunk 3]
```diff
@@ -1,11 +1,13 @@
...
 import { By } from "@angular/platform-browser";
-import { IgxListComponent, IgxListItemComponent, IgxListModule, IgxListPanState } from "./list.component";
+import { IgxListItemComponent } from "./list-item.component";
+import { IgxListPanState } from "./list.common";
+import { IgxListComponent, IgxListModule } from "./list.component";
 
 declare var Simulator: any;
 
-describe("List", () => {
+fdescribe("List", () => {
```
Comment (rkaraivanov) [inline src/list/list.component.spec.ts]: This would only run the List tests. Revert the change.

File: src/list/list.component.ts [hunk 4]
```diff
@@ -1,17 +1,24 @@
...
-    AfterContentInit, Component, ContentChildren, ElementRef, EventEmitter, forwardRef, HostBinding,
-    Inject, Input, NgModule, OnDestroy, OnInit, Output, QueryList, Renderer2, ViewChild
+    Component,
+    ContentChild,
+    ContentChildren,
+    ElementRef,
+    EventEmitter,
+    forwardRef,
+    HostBinding,
+    Inject,
```
Comment (rkaraivanov) [inline src/list/list.component.ts]: `Inject` is not used inside the file. You can remove it from the imports.

**Comment by DDavidkov** [comment]
@kdinev The list-performance sample was fine, but other samples had issues due to the braking changes. They are all fixed now.
